app/server/lib/DocManager.ts

```diff
@@ -733,3 +739,41 @@ export class DocManager extends EventEmitter {
 function extname(fpath: string): string {
   return path.extname("X" + fpath);
 }
+
+async function updateDocumentAttachmentStoreSettingToValidValue(fname: string, provider: IAttachmentStoreProvider) {
+  return updateDocumentSettingsInPlace(fname, (oldSettings) => {
+    const attachmentStoreId = oldSettings?.attachmentStoreId;
+    if (!attachmentStoreId || provider.storeExists(attachmentStoreId)) {
+      return oldSettings;
+    }
+    const newStoreLabel = getConfiguredStandardAttachmentStore();
+    const newStoreId = newStoreLabel && provider.getStoreIdFromLabel(newStoreLabel);
+
+    return {
+      ...oldSettings,
+      attachmentStoreId: newStoreId,
+    };
+  });
+}
+
+// Updates the document's settings (_grist_DocInfo.docSettings) without loading the document.
+async function updateDocumentSettingsInPlace(
+  fname: string,
+  makeChanges: (oldSettings: DocumentSettings | undefined) => DocumentSettings | undefined
+) {
+  const db = await SQLiteDB.openDBRaw(fname, OpenMode.OPEN_EXISTING);
+  const columns = await db.all("PRAGMA table_info(_grist_DocInfo)");
+  if (!columns.some(column => column.name === 'documentSettings')) {
+    return;
+  }
+  const results = await db.all('SELECT id, schemaVersion, documentSettings FROM _grist_DocInfo');
+  const updatePromises = results.map(async (row) => {
+    const settings: DocumentSettings | undefined = safeJsonParse(row.documentSettings, undefined);
+    const newSettings = makeChanges(settings);
+    await db.run('UPDATE _grist_DocInfo SET documentSettings = ? WHERE id = ?',
```

Comment:
It would be nice I think not to run the update if no change is needed, to lower the impact if something goes wrong some day with some new funky addition to `documentSettings`.